python/pyspark/sql/connect/dataframe.py

```diff
@@ -44,7 +44,7 @@
     from pyspark.sql.connect.typing import ColumnOrString, ExpressionOrString
     from pyspark.sql.connect.client import RemoteSparkSession
 
-ColumnOrName = Union[Column, str]
+ColumnOrName = Union[Expression, str]
```

Comment:
is this change needed?

Reply:
I think this fell out of a different PR I had for transform. The reason we need it eventually is because right now from a type perspective select can only use columns but should be able to refer to any expression.

Reply:
```suggestion
ColumnOrName = Union[Column, str]
```

Reply:
tarting mypy annotations test...
annotations failed mypy checks:
python/pyspark/sql/tests/connect/test_connect_basic.py:308: error: Argument 1 to "select" of "DataFrame" has incompatible type "ColumnAlias"; expected "Union[Column, str]"  [arg-type]
python/pyspark/sql/tests/connect/test_connect_basic.py:315: error: Argument 1 to "select" of "DataFrame" has incompatible type "ColumnAlias"; expected "Union[Column, str]"  [arg-type]
F

Reply:
ColumnAlias is an expression and not an unresolved attribtue.